src/rules/no-reactive-literals.ts

```diff
@@ -0,0 +1,69 @@
+import type { TSESTree } from "@typescript-eslint/types"
+import { createRule } from "../utils"
+
+const labeledStatementBase = `SvelteReactiveStatement > ExpressionStatement > AssignmentExpression`
+
+export default createRule("no-reactive-literals", {
+  meta: {
+    docs: {
+      description: "Don't assign literal values in reactive statements",
+      category: "Stylistic Issues",
+      recommended: false,
+      conflictWithPrettier: false,
+    },
+    fixable: "code",
+    schema: [],
+    messages: {
+      noReactiveLiterals: `Do not assign literal values inside reactive statements unless absolutely necessary.`,
+    },
+    type: "suggestion",
+  },
+  create(context) {
+    /**
+     * Reusable method for multiple types of nodes that should warn
+     *
+     * @param node TSESTree.AssignmentExpression the node that was found
+     * @returns void
+     */
+    function warn(node: TSESTree.AssignmentExpression) {
+      // Move upwards to include the entire reactive statement
+      const parent = node.parent?.parent
+
+      if (!parent) {
+        return false
+      }
+
+      const source = context.getSourceCode()
+
+      return context.report({
+        node: parent,
+        loc: parent.loc,
+        messageId: "noReactiveLiterals",
+
+        fix(fixer) {
+          return [
+            // Insert "let" + whatever was in there
+            fixer.insertTextBefore(parent, `let ${source.getText(node)}`),
```

Comment:
I think it's a difficult to make auto-fixes. If the variable has already been declared, you will get a syntax error.

```svelte
<script>
let foo;
$: foo = 42;
</script>
```

We need to check the predefined variables for auto-fix. Or I think it is good not to auto-fix.

Reply:
Swapped to suggestions.